### PR TITLE
Allow non-ASCII symbols in data/save-directory

### PR DESCRIPTION
**Pull request by obligaron**
Contributes to #747

## Background

Windows can per default only access UTF8-Paths with wchar_t function overloads.

References:
- https://stackoverflow.com/questions/396567/is-there-a-standard-way-to-do-an-fopen-with-a-unicode-string-file-path
- https://stackoverflow.com/questions/821873/how-to-open-an-stdfstream-ofstream-or-ifstream-with-a-unicode-filename

## Main Changes (on windows):

- Use `fstream`-Constructor with `wchar_t*` overload
- Definie Unicode for StormLib so that a `wchar_t*` (`TCHAR*`)  `SFileOpenArchive` is present

## Restrictions

Currently config-dir is not supported, cause it's not possible to pass a File-Handle or Stream to `Radon`.
But the game will start anyway, only the ini file is ignored (not read from or written to).

## Tested

- Open Game
- Play Intro
- Create Hero
- Save Hero
- Load Hero
- Delete Hero
- Read only check

## Alternative

An alternative could be https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/use-utf8-code-page
This also works (tested with master and changed manifest) and it's much cleaner. But the main problem is that it's only supported since Windows 10 Version 1903.
If we want to go down this route, I can also open a PR with the adjusted Manifest-File.

### CODE REVIEW

**Comment by AJenbo** [comment]
the alternative seams much more sane, but also windows 10, feels like a steep requirement

**Comment by AJenbo** [comment]
do you have an idea for how to solve the ini issue or should we wait untill implementing https://github.com/diasurgical/devilutionX/issues/1246 ?

**Comment by obligaron** [comment]
> 
> 
> do you have an idea for how to solve the ini issue or should we wait untill implementing #1246 ?

We would need to change the implementation of `Radon` (upstream or local).
simpleini already seems to [support](https://github.com/brofield/simpleini/blob/7bca74f6535a37846162383e52071f380c99a43a/SimpleIni.h#L96) reading/writing from a stream.
So I would wait until #1246 is implemented.

**Review by AJenbo — APPROVED** [review]

File: Source/utils/log.hpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 namespace devilution {
 
 // Local definition to fix compilation issue due to header conflict.
-void app_fatal(const char *pszFmt, ...);
+[[noreturn]] void app_fatal(const char *pszFmt, ...);
```
Comment (AJenbo) [inline Source/utils/log.hpp:14]: thanks

**Comment by AJenbo** [comment]
hmm i get "read only" now when trying to run the game on a Mac

**Pull request merged**